### PR TITLE
Grant Management Webapp RFP

### PR DESCRIPTION
**Pull request by randombishop**
RFP to build a frontend to render grants data for Web3 Foundation, or other grants programs who choose to adopt the same standard.

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for creating the RFP. Generally, it looks good, but could we make it more generic? Our grants program should ideally be just an example/first step. Ideally, such a tool (combining GitHub + website) can benefit other grant programs and on-chain treasuries. It might even make sense to integrate some long-term goals into the RFP, like, for example, some oracles/pallets for treasury integration or a decentralized Github alternative/hosting.

File: docs/RFPs/Open/grant_management_webapp.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+By providing an API, it will also allow to pull the data in a structured way and make it easy to calculate statistics or track different metrics.
+
+Finally, we hope to set Web3 Foundation Grant Program as an example of how other grant programs can leverage a simple yet powerful process to manage their grants.
+Therefore, the web application and the structure of our repositories can be re-used by other grant programs.
+
+
+## Existing prototype
+
+A quick an dirty prototype already exists for the application:
```
Comment (0xCaso) [inline docs/RFPs/Open/grant_management_webapp.md:34]:
```suggestion
A quick and dirty prototype already exists for the application:
```

File: docs/RFPs/Open/grant_management_webapp.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+## Deliverables :nut_and_bolt:
+
+### Grants Page
+
+- Lists grants and their status.
+- Search
+- Filter
+
+For example:
+![screenshot_grants_page](./img/grants_page.png)
```
Comment (0xCaso) [inline docs/RFPs/Open/grant_management_webapp.md:53]:
```suggestion
For example:

![screenshot_grants_page](./img/grants_page.png)
```

File: docs/RFPs/Open/grant_management_webapp.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+### Grants details
+
+- Shows the grant information
+- Overview of the grant project
+- Milestones details and status
+- All documents related to the grant and links to their pull requests (application, deliveries and evaluations)
+
+For example:
+![screenshot_grants_details](./img/grants_details.png)
```
Comment (0xCaso) [inline docs/RFPs/Open/grant_management_webapp.md:64]:
```suggestion
For example:

![screenshot_grants_details](./img/grants_details.png)
```

File: docs/RFPs/Open/grant_management_webapp.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+- Current grant applications.
+- Links to the PR on github, link to the team page to check their history.
+- Status of the application, date opened, number of approvals so far and how many remaining.
+
+
+### Deliveries
+
+- Shows current deliveries pending evaluation.
+- Links to all relevant information and history.
+
```
Comment (0xCaso) [inline docs/RFPs/Open/grant_management_webapp.md:85]:
```suggestion
- Shows current deliveries pending evaluation.
- Links to all relevant information and history.
- Status of the delivery, submission date, evaluator, and if it's on stale.
```

**Comment by randombishop** [comment]
@0xCaso Thanks for the feedback and corrections!

**Comment by randombishop** [comment]
@Noc2 
* What do you mean by more generic? Like applicable to a different github structure and process than current one used by w3f?
* +1 for long term goals, added your ideas as additional notes.
* There's already an indication that the specifics shown in the RFP are just examples to illustrate the product:
```
The features proposed above represent an opinion on what the grant management webapp should do and look like,
but are neither exhaustive nor strict requirements. Teams are welcome to propose their own design and vision for this product.
```
I am afraid if we don't provide concrete examples of what we mean, it will be hard to understand. Using github to manage grants is quite original and probably won't make sense to someone who never saw that process.

File: docs/RFPs/Open/grant_management_webapp.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+## Project Description :page_facing_up: 
+
+The objective of this RFP is a web application that publishes all the information contained in 
+the grants repositories in a way that facilitates navigation for the grant team itself primarily, but also
+to any other interested party.
+
+By providing an API, it will also allow to pull the data in a structured way and make it easy to calculate statistics or track different metrics.
+
+Finally, we hope to set Web3 Foundation Grant Program as an example of how other grant programs can leverage a simple yet powerful process to manage their grants.
+Therefore, the web application and the structure of our repositories can be re-used by other grant programs.
```
Comment (keeganquigley) [inline docs/RFPs/Open/grant_management_webapp.md:29]:
```suggestion
The Web3 Foundation Grants Program is unique in that everything is openly and transparently published on GitHub. As a result of this RFP, we hope the W3F Grants Program can set an example of how other grant programs can leverage a simple yet powerful process to manage their grants.
Therefore, the web application and the structure of our repositories can be re-used by other grant programs.
```

File: docs/RFPs/Open/grant_management_webapp.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+
+## Existing prototype
+
+A quick and dirty prototype already exists for the application:
+- [Backend](https://github.com/w3f/w3f_grants_backend)
+- [Frontend](https://github.com/w3f/w3f_grants_frontend)
+
+These are just a first experiment to test how the app could work, and are completely undocumented,
```
Comment (keeganquigley) [inline docs/RFPs/Open/grant_management_webapp.md:38]:
```suggestion
These examples are just an initial experiment to test how the app could work, and are completely undocumented, 
```

File: docs/RFPs/Open/grant_management_webapp.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+
+## Existing prototype
+
+A quick and dirty prototype already exists for the application:
+- [Backend](https://github.com/w3f/w3f_grants_backend)
+- [Frontend](https://github.com/w3f/w3f_grants_frontend)
+
+These are just a first experiment to test how the app could work, and are completely undocumented, 
+but please feel free to contact [randombishop](nabil@web3.foundation) if you need help trying them out or simply discuss.
```
Comment (keeganquigley) [inline docs/RFPs/Open/grant_management_webapp.md:39]: Should we maybe add grants@web3.foundation here instead of just your personal?

**Comment by randombishop** [comment]
Big thanks @keeganquigley, eagle eye for the details!

**Review by Noc2 — APPROVED** [review]
Looks good to me.

**Review by keeganquigley — APPROVED** [review]

**Comment by Noc2** [comment]
I'm going to merge it, since @randombishop counts as third approval.

**Pull request merged**